### PR TITLE
Move port configuration getter to pf_port.c 

### PR DESCRIPTION
**Pull request by pyhys**
Implement getters and setters for cmina internal data

Add TODOs for existing thread-safety problems

### CODE REVIEW

File: src/device/pf_cmina.h [hunk 1]
```diff
@@ -75,13 +75,40 @@ int pf_cmina_get_file_directory (pnet_t * net, const char ** pp_file_directory);
...
+ */
+int pf_cmina_get_alias (pnet_t * net, const char ** pp_alias);
+
+/**
+ * Store the current alias name of the device.
+ *
+ * TODO: Not yet thread safe
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param alias         In :   The alias name. Terminated string.
```
Comment (olbjo) [inline src/device/pf_cmina.h:106]: Missing tab

**Review by olbjo — APPROVED** [review]
I have reviewed commits 2 and 3. Only one minor comment.
👍

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: src/device/pf_cmina.h (hunk 1 above)
Comment (pyhys) [reply src/device/pf_cmina.h]: Done

File: src/device/pf_port.h [hunk 2]
```diff
@@ -84,6 +84,20 @@ int pf_port_get_next (pf_port_iterator_t * p_iterator);
...
 
+/**
+ * Get port configuration.
+ *
+ * If the local port number is out of range this operation will assert.
+ * NULL will never be returned.
+ *
+ * See also \a pf_port_get_state() for port runtime data.
+ *
+ * @param net              InOut: The p-net stack instance
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_port.h]: Done

[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_port.h]: Done

File: src/device/pf_cmina.h (hunk 1 above)
Comment (pyhys) [reply src/device/pf_cmina.h]: Done

File: src/device/pf_cmina.h [hunk 3]
```diff
@@ -91,6 +118,24 @@ int pf_cmina_get_station_name (pnet_t * net, const char ** pp_station_name);
  */
 int pf_cmina_get_ipaddr (pnet_t * net, pnal_ipaddr_t * p_ipaddr);
 
+/**
+ * Retrieve the current netmask for the device.
+ * @param net              InOut: The p-net stack instance
+ * @param p_netmask        Out:   The netmask.
+ * @return  0  if the operation succeeded.
+ *          -1 if an error occurred.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h]: Refactored
Done

File: src/device/pf_cmina.h [hunk 4]
```diff
@@ -91,6 +118,24 @@ int pf_cmina_get_station_name (pnet_t * net, const char ** pp_station_name);
...
+ */
+int pf_cmina_get_netmask (pnet_t * net, pnal_ipaddr_t * p_netmask);
+
+/**
+ * Retrieve the gateway address.
+ * @param net              InOut: The p-net stack instance
+ * @param p_gateway        Out:   The gateway address.
+ * @return  0  if the operation succeeded.
+ *          -1 if an error occurred.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h]: Refactored
Done

File: src/device/pf_cmina.h [hunk 5]
```diff
@@ -75,13 +75,40 @@ int pf_cmina_get_file_directory (pnet_t * net, const char ** pp_file_directory);
...
+
+/**
+ * Store the current alias name of the device.
+ *
+ * TODO: Not yet thread safe
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param alias         In :   The alias name. Terminated string.
+ * @return  0  if the operation succeeded.
+ *          -1 if an error occurred.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h]: Refactored
Done

File: src/device/pf_cmina.h [hunk 6]
```diff
@@ -75,13 +75,40 @@ int pf_cmina_get_file_directory (pnet_t * net, const char ** pp_file_directory);
...
  */
 int pf_cmina_get_station_name (pnet_t * net, const char ** pp_station_name);
 
+/**
+ * Retrieve the current alias name of the device.
+ *
+ * TODO: Not yet thread safe
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param pp_alias         Out:   The alias name. Terminated string.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h]: Refactored
Done

File: src/device/pf_cmina.h [hunk 7]
```diff
@@ -75,13 +75,40 @@ int pf_cmina_get_file_directory (pnet_t * net, const char ** pp_file_directory);
 
 /**
  * Retrieve the current station name of the device.
+ *
+ * TODO: Not yet thread safe
+ *
  * @param net              InOut: The p-net stack instance
  * @param pp_station_name  Out:   The station name. Terminated string.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h]: Refactored
Done

File: src/device/pf_cmina.c [hunk 8]
```diff
@@ -1237,12 +1240,45 @@ int pf_cmina_get_station_name (pnet_t * net, const char ** pp_station_name)
...
+   *pp_alias = net->cmina_current_dcp_ase.alias_name;
+   return 0;
+}
+
+int pf_cmina_set_alias (pnet_t * net, const char * alias)
+{
+   if (strlen (alias) >= sizeof (net->cmina_current_dcp_ase.alias_name))
+   {
+      return -1;
+   }
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.c]: Done

File: src/device/pf_cmina.c [hunk 9]
```diff
@@ -25,6 +25,9 @@
  *
  * States are SETUP, SET_NAME, SET_IP and W_CONNECT.
  *
+ * Implement setters and getters for net->cmina_current_dcp_ase, so other parts
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.c]: Done

File: src/device/pf_cmina.h [hunk 10]
```diff
@@ -75,13 +75,40 @@ int pf_cmina_get_file_directory (pnet_t * net, const char ** pp_file_directory);
...
+ *
+ * @param net              InOut: The p-net stack instance
+ * @param pp_alias         Out:   The alias name. Terminated string.
+ * @return  0  if the operation succeeded.
+ *          -1 if an error occurred.
+ */
+int pf_cmina_get_alias (pnet_t * net, const char ** pp_alias);
+
+/**
+ * Store the current alias name of the device.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_cmina.h]: Updated docs slightly
Done

File: src/device/pf_cmina.c [hunk 11]
```diff
@@ -1225,28 +1224,58 @@ int pf_cmina_dcp_get_req (
...
 }
 
-int pf_cmina_get_station_name (pnet_t * net, const char ** pp_station_name)
+void pf_cmina_get_station_name (const pnet_t * net, char * station_name)
 {
-   *pp_station_name = net->cmina_current_dcp_ase.station_name;
-   return 0;
+   snprintf (
+      station_name,
+      sizeof (net->cmina_current_dcp_ase.station_name),
```
Comment (fredrikdanielmoller) [inline src/device/pf_cmina.c:1236]:
```suggestion
      PNET_STATION_NAME_MAX_SIZE,
```

File: src/device/pf_cmina.c [hunk 12]
```diff
@@ -1225,28 +1224,58 @@ int pf_cmina_dcp_get_req (
...
 }
 
-int pf_cmina_get_ipaddr (pnet_t * net, pnal_ipaddr_t * p_ipaddr)
+void pf_cmina_get_alias_name (const pnet_t * net, char * alias_name)
 {
-   *p_ipaddr = net->cmina_current_dcp_ase.full_ip_suite.ip_suite.ip_addr;
-   return 0;
+   snprintf (
+      alias_name,
+      sizeof (net->cmina_current_dcp_ase.alias_name),
```
Comment (fredrikdanielmoller) [inline src/device/pf_cmina.c:1245]:
```suggestion
      PF_ALIAS_MAX_SIZE,
```

File: src/device/pf_port.h [hunk 13]
```diff
@@ -84,6 +84,21 @@ int pf_port_get_next (pf_port_iterator_t * p_iterator);
...
+ *
+ * If the local port number is out of range this operation will assert.
+ * NULL will never be returned.
+ *
+ * See also \a pf_port_get_state() for port runtime data.
+ *
+ * @param net              In:    The p-net stack instance
+ * @param loc_port_num     In:    Local port number.
+ *                                Valid range: 1 .. PNET_MAX_PORT
+ * @return Address to port configuration
```
Comment (fredrikdanielmoller) [inline src/device/pf_port.h:98]:
```suggestion
 * @return Port configuration
```

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: src/device/pf_cmina.c (hunk 11 above)
Comment (pyhys) [reply src/device/pf_cmina.c]: Done

File: src/device/pf_cmina.c (hunk 12 above)
Comment (pyhys) [reply src/device/pf_cmina.c]: Done

File: src/device/pf_port.h (hunk 13 above)
Comment (pyhys) [reply src/device/pf_port.h]: Done

**Review by fredrikdanielmoller — APPROVED** [review]

**Pull request merged**